### PR TITLE
Add Support for Hausa Language Bounty #708

### PR DESCRIPTION
**Pull request by Justxd22**
# Duplicate of #728  
 - Created & translated [res/values/strings_ha.arb](https://github.com/Justxd22/cake_wallet/commit/69ad306e51bcb1e8c1e1c65b1f28e8dce6f82f68#diff-977872d4eb9cd77900f1bcbac035bde3fb31f8af2a03c0612512ee2eb99cfecc)
 - Added language in [lib/entities/language_service.dart](https://github.com/Justxd22/cake_wallet/commit/47391059ced066d956884e63124206273af3f5da#diff-08342b16f9e0a48cb8c879d13afa949f0baac91735b4f67cb56811aa69342198)
 - Added language flag [assets/images/flags/hau.png](https://github.com/Justxd22/cake_wallet/commit/47391059ced066d956884e63124206273af3f5da#diff-7242d6010bb66ee18f16bbcbd5eb551f4208a5f199b04c1f79b427d3e8ee162d)
 

Hausa is mainly spoken in northern Nigeria, the Republic of Niger, northern Cameroon, and Ghana, but since #709 i can't use Nigeria flag, therefore Hausa flag is used with hau 639-2/T code

Merging this would close https://github.com/cake-tech/cake_wallet/issues/708

Thanks!

# Pull Request - Checklist  

- [x] Initial Manual Tests Passed

### CODE REVIEW

**Review by OmarHatem28 — CHANGES_REQUESTED** [review]
Still showing the same error

```
======== Exception caught by widgets ===============================================================
The following message was thrown:
Warning: This application's locale, ha, is not supported by all of its localization delegates.

• A CupertinoLocalizations delegate that supports the ha locale was not found.
• A MaterialLocalizations delegate that supports the ha locale was not found.

The declared supported locales for this app are: nl, zh, ar, tr, ru, fr, pl, en, hi, ja, hr, pt, de, it, es, ko, ha, th, uk, my

See https://flutter.dev/tutorials/internationalization/ for more information about configuring an app's locale, supportedLocales, and localizationsDelegates parameters.
```

**Comment by Justxd22** [comment]
Hello @OmarHatem28, the error snippet indicates that 'ha'-'hau' is not officially supported by Flutter according to the  [documentation here](https://api.flutter.dev/flutter/flutter_localizations/GlobalCupertinoLocalizations-class.html). However, the good news is that we can still add it by following the Flutter [Localization docs](https://docs.flutter.dev/development/accessibility-and-localization/internationalization#adding-support-for-a-new-language) and using some extra code. I can help, I've some background in Dart, but since I have never used Flutter before, it will take some time for me to assist in adding support for Hausa, what do you think?

**Comment by Justxd22** [comment]
@OmarHatem28 what do you think should I implement Hausa localization?

**Comment by OmarHatem28** [comment]
> @OmarHatem28 what do you think should I implement Hausa localization?

yeah, I would say go for it, and I know it requires some extra steps to add a non-supported locale, but that's the point of contributing :D you get to add your own code to an open-source project

Looking forward to reviewing your PR when it's ready :)

**Comment by Justxd22** [comment]
@OmarHatem28 can you check if the error is gone, i tried reproducing by building from source, i installed flutter and started building but my build server ran out of space, the workflow built fine tho, if not please provide a way for me to reproduce thanks

**Comment by Justxd22** [comment]
@OmarHatem28 can you review this, thanks

File: lib/utils/ha_intl.dart [hunk 1]
```diff
@@ -0,0 +1,759 @@
...
+  String get unspecifiedDate => 'kwanan wata';
+
+  @override
+  String get unspecifiedDateRange => 'kwanan wata';
+
+  @override
+  String get keyboardKeyAlt => throw UnimplementedError();
+
+  @override
+  String get keyboardKeyAltGraph => throw UnimplementedError();
```
[the remark this answers is not in the document]
Comment (Justxd22) [reply lib/utils/ha_intl.dart:624]: They already have `@override`, please clarify your request

**Comment by Justxd22** [comment]
@OmarHatem28 check if it's fixed, I couldn't find any examples on Cupertino in the docs, Therefore I couldn't add it, i managed to find some code from flutter official repo for Cupertino_en, I don't know if it will work, workflow check passed tho

**Comment by Justxd22** [comment]
@OmarHatem28 how did it go?

**Comment by Justxd22** [comment]
@OmarHatem28 hello can you please test last commit and check if Hausa locale works

**Comment by OmarHatem28** [comment]
@Justxd22 Sorry for the late response, it's a bit busy on our side right now.
I pulled your PR but it still has some errors regarding the `CuprtinoLocalization` probably the code you found is old, as the errors are for new variables that are not in the constructor and for some variables that are no longer supported

**Comment by Justxd22** [comment]
@OmarHatem28 the code i found is copied from [official flutter repo](https://github.com/flutter/flutter/blob/master/packages/flutter_localizations/lib/src/l10n/generated_cupertino_localizations.dart) last updated 25/07/22, I can add the missing ones if you attached the logs or provided some way to locally test my commits before pushing, do you get these errors in logcat?

**Comment by OmarHatem28** [comment]
@Justxd22 the errors are Compile-time errors, so you don't need to be able to run the app to check the errors or logs (although I recommend following the [`howto-build-android.md`](https://github.com/cake-tech/cake_wallet/blob/main/howto-build-android.md) to run the app and try on your end

**Comment by Justxd22** [comment]
@OmarHatem28 does the workflow build contains the error log?

**Comment by OmarHatem28** [comment]
> @OmarHatem28 does the workflow build contains the error log?

Strangely no, maybe it's building an older version of the PR or so, but the Local build gives the error logs for missing Cupertino Import, and then the constructor parameters errors

```
lib/utils/ha_intl.dart:816:9: Error: The superclass, 'GlobalCupertinoLocalizations', has no unnamed constructor that takes no arguments.
  const HaCupertinoLocalizations({
        ^^^^^^^^^^^^^^^^^^^^^^^^
Target kernel_snapshot failed: Exception


FAILURE: Build failed with an exception.
```

**Comment by Justxd22** [comment]
I got the full log, i will try to fix it


```
2023-03-01T19:28:21.9053813Z Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/sys-img2/02 to old ns http://schemas.android.com/sdk/android/repo/sys-img2/01
2023-03-01T19:28:43.2146673Z lib/utils/ha_intl.dart:766:35: Error: Type 'CupertinoLocalizations' not found.
2023-03-01T19:28:43.2193403Z     extends LocalizationsDelegate<CupertinoLocalizations> {
2023-03-01T19:28:43.2199395Z                                   ^^^^^^^^^^^^^^^^^^^^^^
2023-03-01T19:28:43.2205149Z lib/utils/ha_intl.dart:773:10: Error: Type 'CupertinoLocalizations' not found.
2023-03-01T19:28:43.2209879Z   Future<CupertinoLocalizations> load(Locale locale) async {
2023-03-01T19:28:43.2214783Z          ^^^^^^^^^^^^^^^^^^^^^^
2023-03-01T19:28:43.2219112Z lib/utils/ha_intl.dart:962:38: Error: Type 'CupertinoLocalizations' not found.
2023-03-01T19:28:43.2223462Z   static const LocalizationsDelegate<CupertinoLocalizations> delegate =
2023-03-01T19:28:43.2227825Z                                      ^^^^^^^^^^^^^^^^^^^^^^
2023-03-01T19:28:47.2945163Z lib/utils/ha_intl.dart:820:9: Error: The superclass, 'GlobalCupertinoLocalizations', has no unnamed constructor that takes no arguments.
2023-03-01T19:28:47.2946149Z   const HaCupertinoLocalizations({
2023-03-01T19:28:47.2946527Z         ^^^^^^^^^^^^^^^^^^^^^^^^
2023-03-01T19:28:49.6947932Z lib/utils/ha_intl.dart:784:30: Error: 'CupertinoLocalizations' isn't a type.
2023-03-01T19:28:49.6949024Z     return SynchronousFuture<CupertinoLocalizations>(
2023-03-01T19:28:49.7001604Z                              ^^^^^^^^^^^^^^^^^^^^^^
2023-03-01T19:28:49.7002864Z lib/utils/ha_intl.dart:962:38: Error: 'CupertinoLocalizations' isn't a type.
2023-03-01T19:28:49.7003368Z   static const LocalizationsDelegate<CupertinoLocalizations> delegate =
2023-03-01T19:28:49.7003732Z                                      ^^^^^^^^^^^^^^^^^^^^^^
2023-03-01T19:29:12.6998197Z Target kernel_snapshot failed: Exception
2023-03-01T19:29:12.7002922Z 
2023-03-01T19:29:12.8953482Z 
2023-03-01T19:29:12.8953999Z FAILURE: Build failed with an exception.
2023-03-01T19:29:12.8954253Z 
2023-03-01T19:29:12.8954328Z * Where:
2023-03-01T19:29:12.8955131Z Script '/opt/hostedtoolcache/flutter/3.7.5-stable/x64/packages/flutter_tools/gradle/flutter.gradle' line: 1151
2023-03-01T19:29:12.8955441Z 
2023-03-01T19:29:12.8955549Z * What went wrong:
2023-03-01T19:29:12.8955991Z Execution failed for task ':app:compileFlutterBuildRelease'.
2023-03-01T19:29:12.8957834Z > Process 'command '/opt/hostedtoolcache/flutter/3.7.5-stable/x64/bin/flutter'' finished with non-zero exit value 1
2023-03-01T19:29:12.8958141Z 
2023-03-01T19:29:12.8958230Z * Try:
2023-03-01T19:29:12.8958852Z Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
2023-03-01T19:29:12.8959188Z 
2023-03-01T19:29:12.8959344Z * Get more help at https://help.gradle.org
2023-03-01T19:29:12.8959547Z 
2023-03-01T19:29:12.8959657Z BUILD FAILED in 3m 26s
2023-03-01T19:29:13.4995805Z Running Gradle task 'assembleRelease'...                          207.7s
2023-03-01T19:29:13.5078825Z Gradle task assembleRelease failed with exit code 1
2023-03-01T19:29:13.5218688Z ##[error]Process completed with exit code 1.
```

**Comment by Justxd22** [comment]
@OmarHatem28 i think the error is gone now, i'm not sure how i fixed it but it builds fine now i got a build with the language added and working fine, you can try out the [build here](https://github.com/Justxd22/qqtest/suites/12523539749/artifacts/668523708) and [here's the commit](https://github.com/Justxd22/qqtest/commit/5726bf3e4a347a9722fad669323079038fa8d99c)

**Comment by Justxd22** [comment]
@OmarHatem28 can you take a look

**Comment by OmarHatem28** [comment]
Hi @Justxd22 Sorry for the late reply and thanks for updating the PR, unfortunately, it still has errors regarding Cupertino Localization
 
> Strangely no, maybe it's building an older version of the PR or so, but the Local build gives the error logs for missing Cupertino Import, and then the constructor parameters errors
> 
> ```
> lib/utils/ha_intl.dart:816:9: Error: The superclass, 'GlobalCupertinoLocalizations', has no unnamed constructor that takes no arguments.
>   const HaCupertinoLocalizations({
>         ^^^^^^^^^^^^^^^^^^^^^^^^
> Target kernel_snapshot failed: Exception
> 
> 
> FAILURE: Build failed with an exception.
> ```

and here is the error after importing `cupertino`

Screen Shot 2023-05-03 at 12 25 06 AM

**Comment by Justxd22** [comment]
@OmarHatem28 the code in this pr is no longer required, i added `ha_intl.dart` as an attempt to fix the compile errors but now it compiles fine so it's no longer required #914

**Pull request closed without merging**